compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1690,7 +1691,18 @@ private void checkForExportableType(BTypeSymbol symbol, Location pos, HashSet<BT
                 symbolType = Types.getReferredType(symbolType);
                 checkForExportableType(symbolType.tsymbol, pos, visitedSymbols);
                 return;
-            // TODO : Add support for other types. such as union and objects
+            case TypeTags.UNION:
+                ((BUnionType) symbolType).getMemberTypes().forEach(t -> checkForExportableType(t.tsymbol, pos,
+                        visitedSymbols));
+                return;
+            case TypeTags.OBJECT:
+                ((BObjectType) symbolType).fields.values().forEach(f -> checkForExportableType(f.type.tsymbol, pos,
```

Comment:
I think the logic here should be similar to what we have done with record types

Reply:
Updated [9cffdc1](https://github.com/ballerina-platform/ballerina-lang/pull/39970/commits/9cffdc14e7f69d756b3454acb42e91d924499f83)